src/bvals/cc/boundary_communication.cpp

```diff
@@ -105,20 +109,28 @@ TaskStatus SendBoundBufs(std::shared_ptr<MeshData<Real>> &md) {
         const int NtNuNvNkNjNi = Nt * NuNvNkNjNi;
 
         Real threshold = bnd_info(b).var.allocation_threshold;
-        Kokkos::parallel_for(Kokkos::TeamThreadRange<>(team_member, NtNuNvNkNjNi),
-                             [&](const int idx) {
-                               const int t = idx / NuNvNkNjNi;
-                               const int u = (idx % NuNvNkNjNi) / NvNkNjNi;
-                               const int v = (idx % NvNkNjNi) / NkNjNi;
-                               const int k = (idx % NkNjNi) / NjNi + sk;
-                               const int j = (idx % NjNi) / Ni + sj;
-                               const int i = idx % Ni + si;
-
-                               const Real &val = bnd_info(b).var(t, u, v, k, j, i);
-                               bnd_info(b).buf(idx) = val;
-                               if (std::abs(val) >= threshold)
-                                 sending_nonzero_flags(b) = true;
-                             });
+        bool non_zero = false;
+        Kokkos::parallel_reduce(
+            Kokkos::TeamThreadRange<>(team_member, NtNuNvNkNjNi),
+            [&](const int idx, bool &lnon_zero) {
+              const int t = idx / NuNvNkNjNi;
+              const int u = (idx % NuNvNkNjNi) / NvNkNjNi;
+              const int v = (idx % NvNkNjNi) / NkNjNi;
+              const int k = (idx % NkNjNi) / NjNi + sk;
+              const int j = (idx % NjNi) / Ni + sj;
+              const int i = idx % Ni + si;
+
+              const Real &val = bnd_info(b).var(t, u, v, k, j, i);
+              bnd_info(b).buf(idx) = val;
+              if (std::abs(val) >= threshold) {
+                lnon_zero = true;
+              }
```

Comment:
I'm a sucker for replacing if statements when I can
```suggestion
              lnon_zero = lnon_zero || (std::abs(val) >= threshold);
```
assuming `Kokkos::LOr` initializes to `false`, which I think it would.

Reply:
Correct, the logical or is initialized to `0/false`.

Reply:
Given that `lnon_zero` is thread local, we can probably skip the first test and just set
```
lnon_zero = (std::abs(val) >= threshold);
```
don't we?

Reply:
Uh I'm not sure. Is the loop body part of the `join` method for the reducer? I think it's safer and clearer to use `||` to be consistent with expectations for how a `LOr` reduction works.

Reply:
Same as how one wouldn't do `x = 1` for a sum reduction, one would do `x += 1`.

Reply:
I think it should be fine in this case, but agree that it safer and cleaner so I'll go with your suggestion.